e2e/poms/formBuilder.ts

```diff
@@ -0,0 +1,124 @@
+import { Page, expect } from "@playwright/test";
+import { FORMBUILDER_SELECTORS } from "../constants/selectors";
+import { FORMBUILDER_API } from "../constants/selectors/formBuilder";
+
+export default class FormBuilder {
+    constructor(private page: Page) {
+        this.page = page;
+    }
+
+    setFormName = async (formName: string) => {
+        await this.page.getByTestId(FORMBUILDER_SELECTORS.formTitle).click();
+        await this.page.getByTestId(FORMBUILDER_SELECTORS.formRenameTextField).fill(formName);
+        await this.page.getByTestId(FORMBUILDER_SELECTORS.formRenameSubmitButton).click();
+
+        await expect(this.page.getByTestId(FORMBUILDER_SELECTORS.formTitle)).toContainText(formName);
+    };
+
+    addElement = async (elementName: string) => {
+        await this.page.getByRole('button', { name: elementName }).click();
+        await expect(this.page.getByTestId(FORMBUILDER_SELECTORS.questionHeader)).toContainText(elementName);
+    };
+
+    navigateToSubmissionsTab = async () => {
+        await this.page.getByTestId(FORMBUILDER_SELECTORS.submissionsTab).click();
+        await expect(this.page.getByTestId(FORMBUILDER_SELECTORS.submissionsCountLabel)).toBeVisible();
+    };
+
+    navigateToAnalytics = async () => {
+        await this.page.getByTestId(FORMBUILDER_SELECTORS.moreDropdownIcon).click();
+        await this.page.getByTestId(FORMBUILDER_SELECTORS.analyticsMoreTab).click();
+        await expect(this.page.getByTestId(FORMBUILDER_SELECTORS.submissionsInsightsTitle)).toBeVisible();
+    }
+
+    publishForm = async () => await this.page.getByTestId(FORMBUILDER_SELECTORS.publishButton).click();
+
+    returnPublishUrl = async () => {
+        const button = this.page.getByTestId(FORMBUILDER_SELECTORS.publishPreviewButton);
+        await expect(button).toBeEnabled();
+        
+        const href = await button.getAttribute('href');
+        if (!href) {
+            throw new Error('No href attribute found on publish preview button');
+        }
+        return href;
```

Comment:
Throwing an error within a test is not a good practice. It is better to do an assertion with toHaveAttribute before you try to extract the href property so that Playwright can show you meaningful errors.
```suggestion
        await Promise.all([
          expect(button).toBeEnabled(),
          expect(button).toHaveAttribute("href")
        ]);
        
        return button.getAttribute('href');
```